Avoid warnings during global destruction

atoomic opened the pull request:
This is avoiding errors like
```
# Warning was 'Use of uninitialized value in numeric ne (!=) at /usr/local/cpanel/3rdparty/perl/530/lib/perl5/cpanel_lib/Promise/ES6.pm line 418.
```

FGasper commented on lib/Promise/ES6.pm:
```diff
@@ -415,7 +415,7 @@ sub _is_promise {
 
 sub DESTROY {
-    return if $$ != $_[0]{'_pid'};
+    return unless $_[0]{'_pid'} && $$ == $_[0]{'_pid'};
```
Ordinarily I’m averse to `unless`, but here, at least, I can agree it seems at least as clear. <shrug>

FGasper commented:
@atoomic Are you able to come up with a way to test this?

atoomic commented:
a dumb way to reproduce it, would be to mock the object, not sure it's valid steps to reproduce
```
╰─> perl -Ilib -MPromise::ES6 -E 'my $x = bless {}, "Promise::ES6"; undef $x '
Use of uninitialized value in numeric ne (!=) at lib/Promise/ES6.pm line 418.
```

FGasper commented:
@atoomic I’m a bit averse to merging this without reproduction steps. I know it’s been a while, but do you recall if this happened only in testing, or in production as well?

atoomic commented:
I cannot remember if this was coming from testing or other environment,
I think it's always a good idea to add some protection to any logic leaving in DESTROY 

as shown above this can easily be triggered during testing while mocking some functions for example

atoomic committed:
Avoid warnings during global destruction

This can be triggered during unit tests when
mocking the 'new' function for example.

atoomic commented:
I've rebased the PR to match the last refactor

FGasper commented:
> as shown above this can easily be triggered during testing while mocking some functions for example

I wouldn’t say that `bless {}, "Promise::ES6"` constitutes legitimate use of the module, though. (In fact, it’ll be horribly broken now that the actual data structure is an array rather than a hash.) Even in testing, that kind of thing seems more usefully done w/ something like Test::MockObject::Extends.

Anyway, it’s NBD either way. I’ll merge it and add a comment about why it’s there.
